src/cursor.rs

```diff
@@ -0,0 +1,116 @@
+use std::ops::{Add, AddAssign, Index, IndexMut, Sub, SubAssign};
+
+// TODO: AddAssign
+pub struct CursorMut<'a, T> {
+    data: &'a mut [T],
+    index: usize,
+}
+
+impl<'a, T> CursorMut<'a, T> {
+    pub fn new(data: &'a mut [T]) -> Self {
+        CursorMut { data, index: 0 }
+    }
+
+    pub fn as_slice(&self) -> &[T] {
+        &self.data[self.index..]
+    }
+
+    pub fn as_mut_slice(&mut self) -> &mut [T] {
+        &mut self.data[self.index..]
+    }
+
+    pub fn as_ptr(&self) -> *const T {
+        self.as_slice().as_ptr()
+    }
+
+    pub fn as_mut_ptr(&mut self) -> *mut T {
+        self.as_mut_slice().as_mut_ptr()
+    }
+
+    pub fn clone<'b>(&'b mut self) -> CursorMut<'b, T> {
+        CursorMut {
+            data: self.data,
+            index: self.index,
+        }
+    }
+}
+
+impl<'a, T> Add<usize> for CursorMut<'a, T> {
+    type Output = Self;
+
+    fn add(mut self, rhs: usize) -> Self::Output {
+        self.index += rhs;
```

Comment:
If we keep the operator overloads, could you have the non assign impls delegate to the assign ones?  So this would be `self += rhs; self`, for example.

Reply:
Done.